### PR TITLE
HPCC-15663 Fileservices plugin is not threadsafe

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: esp/bindings/http/client/httpclient.cpp [hunk 1]
```diff
@@ -754,7 +755,8 @@ int CHttpClient::postRequest(ISoapMessage &req, ISoapMessage& resp)
 IHttpClientContext* getHttpClientContext()
 {
     static Owned<CHttpClientContext> theHttpClientContext;
-
+    static CriticalSection httpCrit;
```
Comment (ghalliday) [inline esp/bindings/http/client/httpclient.cpp]: more efficient to have this outside the function.  If you're being really clever/modern you could say

static Owned<CHttpClientContext> theHttpClientContext = new CHqttpClientContext();

which is threadsafe on modern compilers.

**Comment by ghalliday** [comment]
@richardkchapman one minor comment

File: esp/bindings/http/client/httpclient.cpp (hunk 1 above)
Comment (richardkchapman) [inline esp/bindings/http/client/httpclient.cpp]: Efficiency was not really a big concern here...

**Comment by richardkchapman** [comment]
@ghalliday Repushed to address minor comment

**Pull request merged**